### PR TITLE
Applitools integration

### PR DESCRIPTION
**Pull request by gnomeontherun**
## PR Checklist

Please check if your PR fulfills the following requirements:

* [x] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [x] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [x] Build related changes
* [x] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

This adds Applitools to our testing setup, and has 3 tests (Toggles, Checkboxes, Colors). It also runs on IE11, Firefox, Chrome, and Edge. It also supports light and dark.

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Other information

This also should setup the GitHub integration where it will block a PR if there is a failure in the visual testing.

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
```
Comment (hippee-lee) [inline cypress/support/commands.js]: Everything is commented out in this file. Do we really want to add it if its not being used yet?

Comment (gnomeontherun) [reply cypress/support/commands.js]: This is a file generated by cypress, I'm ok with removing it since we probably don't need it.

**Comment by bdryanovski** [comment]
I think it will be good to add some kind of small intro documentation into the development guideline. 

* What you need to do to run the visual tests
* Short explanation on the newly added package.json commands
* What to do when Error X ...

**Comment by gnomeontherun** [comment]
There is a test with an intentional change that needs to be reviewed before it can be approved.

**Comment by gnomeontherun** [comment]
The reason for so many gemini test changes is simply the change in how the dev app main-container styles are changed. Prior to the change made in the dev app, gemini nor applitools could scroll and snapshot, but now it can. This means previously if the content was less than the browser size, the screenshot would often snap a much larger page anyways, but if the content was larger than the browser size, it would not snapshot the overflow content. Now it will crop or expand to the content of the page and snapshot.

File: cypress/integration/buttons.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { checkEyes, setup } from '../util';
+
+const test = 'Buttons';
+
+// TODO: Enable these tests again after validation
+export function ButtonSpecs() {
+  xdescribe(test, () => {
```
Comment (hippee-lee) [inline cypress/integration/buttons.js]: Should this suite be skipped?

File: cypress/integration/spec.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import { CheckboxesSpec } from './checkboxes';
+import { ColorSpec } from './color';
+import { ListsSpec } from './lists';
+import { TogglesSpec } from './toggles';
+
+// Organized this way to make one batch for all of the tests in Applitools,
+// otherwise it treats each file as a different batch and makes it hard to
+// see a single run as one unit.
+describe(`Clarity - ${Cypress.env('CLARITY_THEME')}`, () => {
+  // ButtonSpecs();
```
Comment (hippee-lee) [inline cypress/integration/spec.js]: This is commented out but xdescribed in the suite. Not sure what it should be. You may have covered this in dev share and I missed it.

File: scripts/load-vars.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+curl -g https://us-central1-clarity-design-system.cloudfunctions.net/applitools -o .env
```
Comment (hippee-lee) [inline scripts/load-vars.sh]: I know there isn't much we can do but 🙁

File: cypress/integration/buttons.js (hunk 2 above)
Comment (gnomeontherun) [reply cypress/integration/buttons.js]: Yes, I wanted to leave the test suite but not run it for now.

File: cypress/integration/spec.js (hunk 3 above)
Comment (gnomeontherun) [reply cypress/integration/spec.js]: I just commented it out as well to be certain.

File: scripts/load-vars.sh (hunk 4 above)
Comment (gnomeontherun) [reply scripts/load-vars.sh]: Its only meant to run on CI and this is the easiest script to write without writing a longer node script.

**Comment by gnomeontherun** [comment]
> Still lgtm - one thought on the travis.yml. We used to be limited to 5 concurrent jobs but now its off in the settings so adding a 6th job isn't a big deal. If we want to pare down on jobs we could combine the gemini suites into 3 if we had to.

My idea is we'd shuffle a few around to get it down to 5 jobs again, but didn't want to do that yet. Was hopeful we could also remove some gemini tests too.

**Pull request merged**